sdk/core/azure-core/test/ut/uuid_test.cpp

```diff
@@ -127,3 +127,81 @@ TEST(Uuid, validChars)
       uuidKey,
       4);
 }
+
+TEST(Uuid, nilAndDefault)
+{
+  Uuid uuid;
+  ASSERT_TRUE(uuid.IsNil());
+  ASSERT_EQ(uuid.ToString(), "00000000-0000-0000-0000-000000000000");
+  ASSERT_EQ(uuid, Uuid{});
+  ASSERT_EQ(uuid.AsArray(), Uuid::ValueArray({0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0}));
+}
+
+TEST(Uuid, parse)
+{
+  Uuid uuid1 = Uuid::Parse("00112233-4455-6677-8899-aAbBcCdDeEfF");
+
+  ASSERT_FALSE(uuid1.IsNil());
+  ASSERT_EQ(uuid1.ToString(), "00112233-4455-6677-8899-aabbccddeeff");
+  ASSERT_NE(uuid1, Uuid{});
+  ASSERT_EQ(
+      uuid1.AsArray(),
+      Uuid::ValueArray(
+          {0x00,
+           0x11,
+           0x22,
+           0x33,
+           0x44,
+           0x55,
+           0x66,
+           0x77,
+           0x88,
+           0x99,
+           0xAA,
+           0xBB,
+           0xCC,
+           0xDD,
+           0xEE,
+           0xFF}));
+
+  // Empty string
+  ASSERT_THROW(Uuid::Parse(""), std::invalid_argument);
+
+  // Special characters - make sure we're not treating them as byte array
+  ASSERT_THROW(Uuid::Parse("\a\a\a\a\a\a\a\a\a\a\a\a\a\a\a\a"), std::invalid_argument);
+
+  // Spaces before, after, and both.
+  ASSERT_THROW(Uuid::Parse("00000000-0000-0000-0000-000000000000 "), std::invalid_argument);
+  ASSERT_THROW(Uuid::Parse(" 00000000-0000-0000-0000-000000000000"), std::invalid_argument);
+  ASSERT_THROW(Uuid::Parse("00000000-0000-0000-0000-00000000000"), std::invalid_argument);
```

Comment:
I don't see spaces before/after in this test case. Comment/test mismatch?

I am assuming we expect a "zero" Uuid parse to fail?

Reply:
Thanks! See #5943.

Reply:
Btw, isn't it surprising that parsing a nil Uuid fails? Even though we generate it, in the default ctor?

Reply:
That one had one less zero in the last section. I have moved it into a section of its own, and added 4 more possible variants.

Reply:
Okay, great!

I think it's worth having a test for Parse(default ctor.ToString()).

Reply:
I see you added that scenario (in spirit) already. Thanks

Reply:
Yes, we are testing that behavior, maybe not as explicitly, but it is covered. In your future PR, in your future PR, feel free to add a test for `Parse(Uuid{}.ToString())`.